src/NewTools-Debugger/Context.extension.st

```diff
@@ -0,0 +1,20 @@
+Extension { #name : #Context }
+
+{ #category : #'*NewTools-Debugger' }
+Context >> findFailedAssertionContextInSenderChain [
+
+	^ (self findContextSuchThat: [ :ctx | 
+		   (ctx hasTemporaryVariableNamed: #actual) and: [ 
+			   ctx hasTemporaryVariableNamed: #expected ] ]) ifNil: [ 
+		  StDebuggeAssertionFailureContextNotFound new signal ]
+]
+
+{ #category : #'*NewTools-Debugger' }
+Context >> findFailedAssertionResultsInSenderChain [
+	|assertionContext assertionResult|
+	assertionContext := self findFailedAssertionContextInSenderChain.
+	assertionResult := StDebuggerAssertionFailureResult new.
+	assertionResult actual: (assertionContext readVariableNamed: #actual).
+	assertionResult expected: (assertionContext readVariableNamed: #expected).
+	^assertionResult
```

Comment:
> I would probably implement it a bit differently and not relying on variable names:
> 
> ```smalltalk
> Context>>isAssertionMethod
>     ^"better logic here based on selectors (maybe using pragmas to mark assertions)"
> 
> Context>>asAssertionResult
>   "Assertions are implemented with convention to have first argument as an actual value and the second argument as expected result"
>    self isAssertionMethod ifFalse: [ StDebuggeAssertionFailureContextNotFound signal ].
>    ^ StDebuggerAssertionFailureResult actual: self arguments first expected: self arguments second
> ```
> 
> And then the find method will be more simple and readable:
> 
> ```smalltalk
> Context>> findFailedAssertionResultsInSenderChain
>      assertionContext := self findContextSuchThat: [ :ctx | ctx isAssertionMethod ].
>      assertionContext ifNil: [ ^  StDebuggeAssertionFailureContextNotFound signal ].
>      ^assertionContext asAssertionResult
> ```

Thanks for the suggestion, it would be better by marking the assertion methods with pragmas and then accessing the arguments.
 I will open an issue.